src/app/[lng]/page.tsx

```diff
@@ -1,36 +1,70 @@
+'use client';
+import { cookieName } from '../i18n/settings';
 import { postReissue } from '@/apis/auth';
 import { AUTH_KEYS } from '@/constants/token';
+import { useDidMount } from '@/hooks/common/useDidMount';
 import { getTokenFromCookie } from '@/utils/auth/tokenController';
-import { redirect } from 'next/navigation';
+import { setLocalCookie } from '@/utils/cookieController';
+import { afterDay60 } from '@/utils/date';
+import { useRouter } from 'next/navigation';
 import { NextResponse } from 'next/server';
+import { useCallback } from 'react';
 
-export default async function Home() {
-  const { accessToken, refreshToken } = (await getTokenFromCookie()) as {
-    accessToken: string;
-    refreshToken: string;
-  };
-
-  if (accessToken && refreshToken) return redirect('/grouping');
-
-  try {
-    const {
-      token: { accessToken: reIssuedAccessToken, refreshToken: reIssuedRefreshToken },
-    } = await postReissue(
-      { accessToken, refreshToken },
-      { headers: { 'X-AUTH-TOKEN': accessToken } }
-    );
-
-    const response = NextResponse.next();
-    response.cookies.set(AUTH_KEYS.accessToken, reIssuedAccessToken, {
-      expires: new Date(Date.now() + 1000 * 60 * 60 * 24 * 60 + 9 * 60 * 60 * 1000),
-    });
-    response.cookies.set(AUTH_KEYS.refreshToken, reIssuedRefreshToken, {
-      expires: new Date(Date.now() + 1000 * 60 * 60 * 24 * 60 + 9 * 60 * 60 * 1000),
-    });
-
-    redirect('/grouping');
-  } catch (e) {
-    console.log(e);
-    return redirect('/join');
-  }
+export default function Home() {
+  const router = useRouter();
+
+  const listenRN = useCallback(() => {
+    const listener = (event: any) => {
+      const { data, type } = JSON.parse(event.data);
+      if (data === 'ko')
+        setLocalCookie(cookieName, 'ko', {
+          expires: afterDay60,
+        });
+      else
+        setLocalCookie(cookieName, 'en', {
+          expires: afterDay60,
+        });
+    };
```

Comment:
이렇게 하면 더 간결해질 것 같아요!

```suggestion
        setLocalCookie(cookieName, data, {
          expires: afterDay60,
        });
```